chcemvediet/apps/inforequests/tests/test_models/test_inforequest.py

```diff
@@ -1006,7 +1104,41 @@ def test_closed_and_not_closed_query_methods(self):
         result = Inforequest.objects.not_closed()
         self.assertItemsEqual(result, [inforequest2, inforequest4])
 
+    def test_successful_and_unsuccessful_query_methods(self):
+        Inforequest.objects.all().delete()
+        successful_inforequests = []
+        unsuccessful_inforequests = []
+        for closed in (False, True):
+            for disclosure_level in Action.DISCLOSURE_LEVELS._inverse.keys():
+                inforequest, _, _ = self._create_inforequest_scenario(
+                    {u'closed': closed},
+                    (u'disclosure', dict(disclosure_level=disclosure_level)),
+                )
+                if closed:
+                    if disclosure_level >= Action.DISCLOSURE_LEVELS.PARTIAL:
+                        successful_inforequests.append(inforequest)
+                    else:
+                        unsuccessful_inforequests.append(inforequest)
+        self.assertItemsEqual(Inforequest.objects.successful(), successful_inforequests)
+        self.assertItemsEqual(Inforequest.objects.unsuccessful(), unsuccessful_inforequests)
```

Comment:
V teste je lepsie spravit explicitnu tabulku: closed / disclosure level / successful. Aby sme logiku rozhodovania v teste neimplementovali, ale mali to uz dopredu rucne rozhodnute a iba overili.

Reply:
:ok: Chcel som si to najprv zjednodusit, ale iba som si to skomplikoval. Nakoniec som nespravil tabulku ale ulozil som si inforequesty do premennych, analogicky ako pri ostatnych query metodach.